pkg/resources/manager/manager.go

```diff
@@ -197,6 +221,10 @@ func (r *resourceManager) initialize(ctx context.Context, machines []*api.Machin
 			return err
 		}
 
+		for _, value := range resources.UnsortedList() {
+			r.registredResources[value] = s
```

Comment:
If a resource is managed by multiple sources, it will be overwritten without warning. Ideally, an error should be thrown in such cases.

Reply:
we have this logic on line 233. This part of code will need to refactor.